src/loader/SpineResource.ts

```diff
@@ -1,28 +1,42 @@
-import { SkeletonData } from "@esotericsoftware/spine-core";
+import { AnimationStateData, Skeleton, SkeletonData } from "@esotericsoftware/spine-core";
 import { Engine, ReferResource, Texture2D } from "@galacean/engine";
-import { SpineAnimationRenderer } from "../SpineAnimationRenderer";
 
-export class SkeletonDataResource extends ReferResource {
-  readonly textures: Texture2D[] = [];
+export class SpineResource extends ReferResource {
+  private static _cache = new Map<SkeletonData, SpineResource>();
+  private _texturesInSpineAtlas: Texture2D[] = [];
   private _skeletonData: SkeletonData;
+  private _animationData: AnimationStateData;
 
-  get skeletonData() {
+  /** @internal */
+  static addRefCount(skeletonData: SkeletonData, count: number) {
+    const resource = SpineResource._cache.get(skeletonData);
+    // @ts-ignore
+    resource._addReferCount(count);
+  }
```

Comment:
Reference counting should not be specialized, use Templet design by `_markAsTemplate`

Reply:
done. Already delete this function.